include/mxnet/tensor_blob.h

```diff
@@ -148,6 +148,43 @@ class TBlob {
   TBlob(const mshadow::Tensor<Device, dim, DType> &src) {  // NOLINT(*)
     *this = src;
   }
+
+  /*!
+   * \brief constructor from contiguous memory
+   * \param dptr data pointer
+   * \param shape data shape
+   * \param stream device stream
+   * \tparam Device which device the tensor is on
+   * \tparam dim tensor dimension
+   * \tparam DType the type of elements in the tensor
+   */
+  template<typename Device, int dim, typename DType>
+  TBlob(DType *dptr, const mshadow::Shape<dim> &shape,
+        mshadow::Stream<Device> *stream) {
```

Comment:
`TBlob` constructor is not expected to receive a stream since it's not supposed to be a tensor type for devices. The existing constructors should suffice any use cases.